src/components/FlyoutMenu/FlyoutMenu.test.jsx

```diff
@@ -0,0 +1,153 @@
+import React from 'react';
+import '@testing-library/jest-dom';
+import { waitForElement, render, waitFor } from '@testing-library/react';
+
+import FlyoutMenu from './FlyoutMenu';
+
+const applyButtonText = 'Apply';
+const cancelButtonText = 'Cancel';
+
+describe('FlyoutMenu', () => {
+  it('Renders an open transactional menu', async () => {
+    const { queryByText } = render(<FlyoutMenu open transactional triggerId="flyout-test" />);
+
+    const applyButtonTest = await waitForElement(() => queryByText(applyButtonText));
+
+    const cancelButtonTest = await waitForElement(() => queryByText(cancelButtonText));
+
+    expect(applyButtonTest).toBeTruthy();
+    expect(cancelButtonTest).toBeTruthy();
+  });
+
+  it('Renders an open menu', async () => {
+    const { queryByText } = render(<FlyoutMenu open triggerId="flyout-test" />);
+
+    const applyButtonTest = queryByText(applyButtonText);
+
+    const cancelButtonTest = queryByText(cancelButtonText);
+
+    expect(applyButtonTest).toBeNull();
+    expect(cancelButtonTest).toBeNull();
+  });
+
+  it('Renders closed', async () => {
+    const { queryByText } = render(<FlyoutMenu triggerId="flyout-test" />);
+
+    const applyButtonTest = await waitFor(() => queryByText(applyButtonText));
+
+    const cancelButtonTest = await waitFor(() => queryByText(cancelButtonText));
+
+    expect(applyButtonTest).toBeNull();
+    expect(cancelButtonTest).toBeNull();
+  });
+
+  it('Cancel Action', async () => {
+    const cancelAction = jest.fn();
+
+    const { queryByText } = render(
+      <FlyoutMenu open triggerId="flyout-test" transactional onCancel={cancelAction} />
+    );
+
+    const cancelButton = await waitForElement(() => queryByText(cancelButtonText));
+
+    cancelButton.click();
```

Comment:
What `click()` is being used here? Usually I see [`fireEvent.click(cancelButton)`](https://testing-library.com/docs/react-testing-library/cheatsheet#events) - is this just a proxy for the same thing?